packages/components/src/components/editor/condition-builder/ConditionGroup.tsx

```diff
@@ -0,0 +1,66 @@
+import { IvyIcons } from '@axonivy/ui-icons';
+import { Condition, type ConditionProps } from './Condition';
+import { useConditionBuilderContext, type LogicalOperator } from './ConditionBuilderContext';
+import { BasicSelect } from '@/components/common/select/select';
+import { Flex } from '@/components/common/flex/flex';
+import { Label } from '@radix-ui/react-label';
+import { Button } from '@/components/common/button/button';
+
+export interface ConditionGroup {
+  conditions: Condition[];
+  logicalOperator: LogicalOperator;
+}
+
+interface ConditionGroupProps extends Pick<ConditionProps, 'groupIndex'> {
+  group: ConditionGroup;
+  groupCount: number;
+}
+
+export const ConditionGroup = ({ group, groupIndex, groupCount }: ConditionGroupProps) => {
+  const { updateLogicalOperator, addCondition, removeConditionGroup, conditionMode, logicalOperatorOptions } = useConditionBuilderContext();
+
+  return (
+    <Flex direction='column' gap={2} className='condition-builder__group'>
+      <Flex
+        direction='column'
+        style={
+          conditionMode === 'nested-condition'
+            ? { border: 'var(--basic-border', borderRadius: 'var(--border-r2)', padding: 'var(--size-2)' }
+            : undefined
+        }
```

Comment:
I think it is sometimes fine to use inline styles in the impls, but here I think we should do all the same way, via a separate CSS file, and use the known variables for this variables :)
See the other components for how.

And stuff like the condition mode should be a simple data argument, so you can adjust the styles in the CSS file according to it:
`data-condition-mode={conditionMode}`